### PR TITLE
feat(tx-pool): Add new TransactionPool trait for getting the best transactions with blob fee

### LINKED ISSUE #4822: Add new TransactionPool trait for getting the best transactions with blob fee
### Describe the feature

we currently have to check against the internally enforced basefee

https://github.com/paradigmxyz/reth/blob/66fc1964d7aed37df56f397a7a234e55da27d4ef/crates/transaction-pool/src/traits.rs#L229-L236

which checks fetches the best transactions accordingly:

https://github.com/paradigmxyz/reth/blob/66fc1964d7aed37df56f397a7a234e55da27d4ef/crates/transaction-pool/src/pool/txpool.rs#L227-L245

with upcoming blob fee, we need the same for blob fee, since the pool's tracked blob fee could be different then the targeted (for example block building)

## TODO
* Add new function `best_transactions_with_attributes` which takes a new struct that has: `basefee: u64, blob_fee: Option<u64>`
 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by PanGan21**
Closes: #4822

### CODE REVIEW

File: crates/transaction-pool/src/pool/txpool.rs [hunk 1]
```diff
@@ -245,6 +245,39 @@ impl<T: TransactionOrdering> TxPool<T> {
         }
     }
 
+    /// Returns an iterator that yields transactions that are ready to be included in the block with
+    /// the given base fee and optional blob fee.
+    pub(crate) fn best_transactions_with_attributes(
```
Comment (PanGan21) [inline crates/transaction-pool/src/pool/txpool.rs:250]: Hey @mattsse  please let me know your opinion about this function and then I can also write the related unit tests

File: crates/transaction-pool/src/lib.rs [hunk 2]
```diff
@@ -419,6 +419,14 @@ where
         self.pool.best_transactions_with_base_fee(base_fee)
     }
 
+    fn best_transactions_with_attributes(
+        &self,
+        base_fee: u64,
+        blob_fee: Option<u64>,
```
Comment (mattsse) [inline crates/transaction-pool/src/lib.rs:425]: I want a new struct type for this, this way it's easier to call and it won't break if we need more attributes in the future

File: crates/transaction-pool/src/pool/blob.rs [hunk 3]
```diff
@@ -76,6 +76,14 @@ impl<T: PoolTransaction> BlobTransactions<T> {
         Some(tx)
     }
 
+    pub(crate) fn satisfy_attributes(
+        &self,
+        basefee: u64,
+        blob_fee: Option<u64>,
+    ) -> Vec<Arc<ValidPoolTransaction<T>>> {
+        unreachable!()
```
Comment (mattsse) [inline crates/transaction-pool/src/pool/blob.rs:84]:
```suggestion
        Vec::new()
```

**Review by mattsse — CHANGES_REQUESTED** [review]
this is great, ty.

one suggestion to turn the arguments into a new struct type

File: crates/transaction-pool/src/lib.rs (hunk 2 above)
Comment (PanGan21) [reply crates/transaction-pool/src/lib.rs]: I added `BestTransactionsAttributes` struct as the argument

File: crates/transaction-pool/src/pool/blob.rs (hunk 3 above)
Comment (PanGan21) [reply crates/transaction-pool/src/pool/blob.rs]: Fixed

**Review by mattsse — APPROVED** [review]
this is great, tysm!

**Pull request merged**